doc/source/placement-group.rst

```diff
@@ -0,0 +1,140 @@
+- **Resource reservation in multi-tenant cluster**: You can reserve resources for your job ahead of time, so that you won't need to worry that resource competition slows down your job.
+
+.. note::
+
+  Ray's placement group is still under active development. Please try out and give us feedback and report bugs!
+  APIs are subject to change in the future.
+
+Quick Start
+-----------
+
+Let's see an example of using placement group. Note that this example is done within a single node.
+
+.. code-block:: python
+  
+  import ray
+  from pprint import pprint
+  # Import placement group APIs.
+  from ray.experimental.placement_group import (
+      placement_group,
+      placement_group_table,
+      remove_placement_group
+  )
+
+  ray.init(num_gpus=2, resources={"TPU": 2})
+
+  # Bundle is a dictionary of resource requirements.
+  # Bundles are reserved on Ray clusters according to placement strategy.
+  # Tasks or actors can be scheduled on each bundle.
+  gpu_bundle = {"GPU": 2}
+  tpu_bundle = {"TPU": 2}
+  
+  # Reserve 2 of GPU bundles with strict pack strategy.
+  # It means Ray will reserve 2 GPU resources on the same node (strict pack) within a Ray cluster.
+  # Using this placement group for scheduling actors or tasks will guarantee that they will
+  # collocate in the same node.
+  pg = placement_group([gpu_bundle, tpu_bundle], strategy="STRICT_PACK")
+  
+  # Wait until placement group is created.
+  ray.get(pg.ready())
+
+  # You can also use ray.wait.
+  ready, unready = ray.wait([pg.ready()], timeout=0)
+
+  # You can look at placement group states using this API.
+  pprint(placement_group_table(pg))
+
+  # Now let's define actors that uses GPUs.
+  @ray.remote(num_gpus=1)
+  class GPUActor:
+      def __init__(self):
+          pass
+
+  # Now let's define actors that uses TPUs.
+  @ray.remote(resources={"TPU": 1})
+  class TPUActor:
+      def __init__(self):
+          pass
+
+  # Create GPU actors on a gpu bundle.
+  gpu_actors = [GPUActor.options(
+          placement_group=pg,
+          placement_group_bundle_index=0) # Index of gpu_bundle is 0.
+      .remote() for _ in range(2)]
+
+  # Create TPU actors on a tpu bundle.
+  tpu_actors = [GPUActor.options(
+          placement_group=pg,
+          placement_group_bundle_index=0) # Index of gpu_bundle is 0.
+      .remote() for _ in range(2)]
+
+  # Now, you can guarantee all gpu and tpu actors are located on the same node
+  # because they are scheduled on a placement group that has STRICT_PACK strategy.
+  # You can see this by checking all actors have the same node id.
+  pprint(ray.actors())
+
+  # Remove the placement group since we don't use it anymore.
+  # This API is asynchronous.
+  remove_placement_group(pg)
+
+  # Wait until placement group is killed.
+  import time
+  time.sleep(1)
+  # Check the placement group has died.
+  pprint(placement_group_table(pg))
+
+  ray.shutdown()
```

Comment:
nice example though i had a hard time following - maybe key concepts and diagrams would help a lot!

Reply:
I think a state diagram / flowchart would work nicely here. It would be good to show the stages of a placement group - created, scheduling, in use by tasks/actors, etc.

Reply:
Oh, now I'm seeing the Lifecycle section below, oops! I think that would work well as a flowchart, though :)

Reply:
I will add state diagram, but I think we can merge this PR first!